config/eventvalidation.go

```diff
@@ -0,0 +1,138 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	validator "github.com/asaskevich/govalidator"
+)
+
+// VASTEventElement indicates valid VAST event element
+type VASTEventElement string
+
+const (
+	ImpressionVASTElement             VASTEventElement = "impression"
+	TrackingVASTElement               VASTEventElement = "tracking"
+	ClickTrackingVASTElement          VASTEventElement = "clicktracking"
+	CompanionClickThroughVASTElement  VASTEventElement = "companionclickthrough"
+	ErrorVASTElement                  VASTEventElement = "error"
+	NonLinearClickTrackingVASTElement VASTEventElement = "nonlinearclicktracking"
+)
+
+var vastEventElementMap = map[VASTEventElement]struct{}{
+	ImpressionVASTElement:             {},
+	TrackingVASTElement:               {},
+	ClickTrackingVASTElement:          {},
+	CompanionClickThroughVASTElement:  {},
+	ErrorVASTElement:                  {},
+	NonLinearClickTrackingVASTElement: {},
+}
+
+// TrackingEventType indicates quartile events
+type TrackingEventType string
+
+const (
+	Start         TrackingEventType = "start"
+	FirstQuartile TrackingEventType = "firstQuartile"
```

Comment:
Yes. These values are case-sensitive as per [IAB VAST Validator](https://vastvalidator.iabtechlab.com/dash) 

If you try following `Sample VAST Tag with version 4.2` containing `firstquartile`  (invalid), which suppose to be `firstQuartile`, validator shows following validation. It means it will result in invalid VAST

**NOTE**: Ensure VAST Version 4.2 is selected there, on the tool screen
<img width="1282" alt="image" src="https://user-images.githubusercontent.com/9314645/154609818-0e9afbca-4e47-480e-8345-0539377198a1.png">


`Sample VAST Tag with version 4.2`
```xml
<VAST version="4.2" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns="http://www.iab.com/VAST">
  <Ad id="20001" sequence="1">
    <InLine>
      <AdSystem version="1">iabtechlab</AdSystem>
      <Error><![CDATA[https://example.com/error]]></Error>
      <Extensions>
        <Extension type="iab-Count">
          <total_available>
            <![CDATA[ 2 ]]>
          </total_available>
        </Extension>
      </Extensions>
      <Impression id="Impression-ID"><![CDATA[https://example.com/track/impression]]></Impression>
      <Pricing model="cpm" currency="USD">
        <![CDATA[ 25.00 ]]>
      </Pricing>
      <AdServingId>a532d16d-4d7f-4440-bd29-2ec0e693fc80</AdServingId>
      <AdTitle>iabtechlab video ad</AdTitle>
      <Creatives>
        <Creative id="5480" sequence="1" adId="2447226">
          <Linear>
            <TrackingEvents>
              <Tracking event="start" ><![CDATA[https://example.com/tracking/start]]></Tracking>
              <Tracking event="progress" offset="00:00:10"><![CDATA[http://example.com/tracking/progress-10]]></Tracking>
              <Tracking event="firstquartile"><![CDATA[https://example.com/tracking/firstQuartile]]></Tracking>
              <Tracking event="midpoint"><![CDATA[https://example.com/tracking/midpoint]]></Tracking>
              <Tracking event="thirdQuartile"><![CDATA[https://example.com/tracking/thirdQuartile]]></Tracking>
              <Tracking event="complete"><![CDATA[https://example.com/tracking/complete]]></Tracking>
            </TrackingEvents>
            <Duration>00:00:16</Duration>
            <MediaFiles>
              <MediaFile id="5241" delivery="progressive" type="video/mp4" bitrate="2000" width="1280" height="720" minBitrate="1500" maxBitrate="2500" scalable="1" maintainAspectRatio="1" codec="H.264">
                <![CDATA[https://iab-publicfiles.s3.amazonaws.com/vast/VAST-4.0-Short-Intro.mp4]]>
              </MediaFile>
              <MediaFile id="5244" delivery="progressive" type="video/mp4" bitrate="1000" width="854" height="480" minBitrate="700" maxBitrate="1500" scalable="1" maintainAspectRatio="1" codec="H.264">
                <![CDATA[https://iab-publicfiles.s3.amazonaws.com/vast/VAST-4.0-Short-Intro-mid-resolution.mp4]]>
              </MediaFile>
              <MediaFile id="5246" delivery="progressive" type="video/mp4" bitrate="600" width="640" height="360" minBitrate="500" maxBitrate="700" scalable="1" maintainAspectRatio="1" codec="H.264">
                <![CDATA[https://iab-publicfiles.s3.amazonaws.com/vast/VAST-4.0-Short-Intro-low-resolution.mp4]]>
              </MediaFile>
            </MediaFiles>
            <VideoClicks>
              <ClickThrough id="blog">
                <![CDATA[https://iabtechlab.com]]>
              </ClickThrough>
            </VideoClicks>
          </Linear>
          <UniversalAdId idRegistry="Ad-ID">8465</UniversalAdId>
          <UniversalAdId idRegistry="Foo-ID">4444323</UniversalAdId>
        </Creative>
      </Creatives>
    </InLine>
  </Ad>
</VAST>
```

Validation Errors :
```sh
Errors:
Line:Col[26:47]:cvc-enumeration-valid: Value 'firstquartile' is not facet-valid with respect to enumeration '[mute, unmute, pause, resume, rewind, skip, playerExpand, playerCollapse, loaded, start, firstQuartile, midpoint, thirdQuartile, complete, progress, closeLinear, creativeView, acceptInvitation, adExpand, adCollapse, minimize, close, overlayViewDuration, otherAdInteraction]'. It must be a value from the enumeration.
Line:Col[26:47]:cvc-attribute.3: The value 'firstquartile' of attribute 'event' on element 'Tracking' is not valid with respect to its type, 'null'.
```